### DISCUSSION TITLE
Scripts.V3 as npm CLI

### DISCUSSION THREAD

**Discussion by ross-lanes** [Azure/api-management-developer-portal#1612] [General]
It would be awesome if you all would publish this as an npm package, adding the needed values (bin) to package.json and expositing a main script file so we could use the scripts in our ADO Pipelines/GitHub Actions.  

Unless I have miss understood, right now, we need to fork/clone this repo to use the scripts for an automated migration process.  

If this repo was published as a NPM Package with Scripts.V3 being exposed as a node CLI we could just do the following:

```bash
npx api-managment-developer-portal migrate --sourceSubscriptionId "< your subscription ID >" --sourceResourceGroupName "< your resource group name >" --sourceServiceName "< your service name >" --destSubscriptionId "< your subscription ID >" --destResourceGroupName "< your resource group name >" --destServiceName "< your service name >"
```

Edit:  ops I messed up the example command, corrected

**Comment by mikebudzynski** [comment]
Thanks for feedback, @Bestadd. In the future, we plan to incorporate portal's automation functionality into the existing API Management's tooling - PowerShell and Azure CLI modules.

**Reply by ross-lanes** [reply]
Thank you for the quick reply.

Is there an issue I can watch so I know when the APIM tooling is updated to include dev portal automation?  

Also, was my understanding correct, the preferred method to use the scripts in an automated fashion is to fork the repo and attach your own ADO Pipelines or GitHub Actions? The Pipeline documentation doesn't exilically say fork.

**Reply by mikebudzynski** [reply]
> Is there an issue I can watch so I know when the APIM tooling is updated to include dev portal automation?

I don't recall an Issue to monitor this improvement - feel free to open a new one. We don't have precise timelines yet, but it's unlikely this it will be implemented in the next 6 months.

> Also, was my understanding correct, the preferred method to use the scripts in an automated fashion is to fork the repo and attach your own ADO Pipelines or GitHub Actions? The Pipeline documentation doesn't exilically say fork.

We do not prescribe any service for automation - you can use whichever solution works for you or whichever solution is already in place. The scripts rarely change, but the latest production-ready version is always located in the latest ["release" on GitHub](https://github.com/Azure/api-management-developer-portal/releases). We do not recommend using scripts from the `main` branch, they may be untested. You can either dynamically fetch the script from the repository as part of the DevOps process or just create a static task with a script.
